### DISCUSSION TITLE
Docstrings

### DISCUSSION THREAD

**Discussion by plane** [racket/rhombus#175] [Ideas]
Rhombus could be an opportunity to incorporate docstrings into the syntax.  I don't really know what it would look like, but the least ambitious version could be pretty simple:

```
fun fahrenheit_to_celsius(f):
  "Convert a temperature from degrees Fahrenheit to degrees Celsius"
  (f - 32) * 5/9
```

Docstrings like this are commonly used in a lot of languages including popular languages like Python, so having them in Rhombus might be familiar and welcome to people who are used to those languages.

More ambitious ideas for docstrings could include integration/interaction with:
- Scribble (not sure how this would look)
- DrRacket (show docstrings when you mouse over a function?)
- at-exp, when this is incorporated into Rhombus - something like `@doc{"..."}`?

I don't really have any more specific ideas than what I've put here.  Any thoughts?

**Comment by sorawee** [comment]
See also #60

**Comment by mflatt** [comment]
Maybe it goes without saying, but I think `@` notation and #60 is the direction to try. I've added `@` notation support to the shubbery proposal (#122) and implementation.

**Reply by tgbugs** [reply]
`@` notation is also useful for being able to embed arbitrary syntax as @jeapostrophe does in remix. This is useful for writing single files that contain otherwise incompatible syntax. I haven't kept a close enough eye on what other facilities shrubbery syntax provides for enabling blocks of arbitrary syntax, but `@` expressions are a generalized solution.

**Reply by rocketnia** [reply]
I really like that incorporation of `@` syntax into the shrubbery notation. Where the original has s-expression arguments like `@foo[a b (+ c d)]`, having comma-separated shrubbery expressions like `@foo[a, b, c + d]` feels like a very straightforward adaptation.

**Reply by hendrikboom3** [reply]
A piece of history: The original Lisp 1.5 manual (anyone remember that?) said that commas were equivalent to paces.  It's no longer true (if it ever was; I never got to use the original Lisp 1.5 but used it as a reference for my own implemmentations), but it brings back memories.  I remember the smell of the original manual as a new book was quite distinctive.

**Reply by benknoble** [reply]
@hendrikboom3 Clojure does that, too, but I didn't know it originated that far back!
